src/libs/actions/Report.js

```diff
@@ -425,6 +417,49 @@ function removeOptimisticActions(reportID) {
     });
 }
 
+/**
+ * Retrieve an iouReport, but do not associate it with the chatReport. For example, the user could be viewing a paid
+ * report, and we must never override the chatReport's 'iouReportID' value with a paid report ID. Doing so would cause
+ * 'hasOutstandingIOU' to be set to false, preventing the IOUPreview and IOUBadge Components from displaying.
+ *
+ * @param {Number} iouReportID - ID of the report we are fetching
+ * @param {Number} chatReportID - associated chatReportID which should be added to IOU report data
+ */
+function fetchIOUReportByID(iouReportID, chatReportID) {
+    fetchIOUReport(iouReportID, chatReportID)
+        .then(iouReportObject => setLocalIOUReportData(iouReportObject))
+        .catch(error => console.error(`Error fetching IOU Report ${iouReportID}: ${error}`));
+}
+
+/**
+ * Retrieve an iouReport, associating it with a chatReport. This should be called in place of `fetchIOUReportByID`
+ * when we believe that the iouReport is open and currently linked to the chatReport.
```

Comment:
NAB  it seems like the primary distinguishing feature between  `fetchIOUReportByIDAndUpdateChatReport` and `fetchIOUReportByID` is that `fetchIOUReportByIDAndUpdateChatReport` should _never_ be used when the IOU report being fetched is closed. Maybe it would be clearer to have something like `fetchOpenIOUReport` and `fetchIOUReportByID`?

Reply:
That's correct and a good idea, but there is an exception: When opening the details modal, rather than distinguishing between open and closed chat reports we will always retrieve the updated report data, without updating the report asociation. I would like to improve this by selectively calling `fetchIOUReportByIDAndUpdateChatReport` or perhaps `fetchAndUpdateActiveIOUReport` for reports that are currently listed as the active IOU in the chatReport Onyx data -- but this isn't part of the plan for launch. 

I like that the naming is simpler, but it doesn't quite get the point across that one function will lead to the chatReport being updated, and one will not.

Reply:
Resolving this, please let me know if you have any other thoughts or concerns here.